packages/page-accounts/src/Sidebar/AccountMenuButtons.tsx

```diff
@@ -0,0 +1,141 @@
+  const { t } = useTranslation();
+  const [isTransferOpen, toggleIsTransferOpen] = useToggle();
+
+  const _onForgetAddress = useCallback(
+    (): void => {
+      onForgetAddress();
+      onUpdateName && onUpdateName();
+    },
+    [onForgetAddress, onUpdateName]
+  );
+
+  const toggleIsEditing = useCallback(() => {
+    flags.isEditable && toggleIsEditingName();
+    toggleIsEditingTags();
+  }, [flags.isEditable, toggleIsEditingName, toggleIsEditingTags]);
+
+  const _onUpdateName = useCallback(
+    (): void => {
+      onSaveName();
+      onUpdateName && onUpdateName();
+    },
+    [onSaveName, onUpdateName]
+  );
+
+  const updateName = useCallback(() => {
+    if (isEditingName && (flags.isInContacts || flags.isOwned)) {
+      _onUpdateName();
+      toggleIsEditingName();
+    }
+  }, [isEditingName, flags.isInContacts, flags.isOwned, _onUpdateName, toggleIsEditingName]);
+
+  const onEdit = useCallback(() => {
+    if (isEditing) {
+      updateName();
+      onSaveTags();
+    }
+
+    toggleIsEditing();
+  }, [isEditing, toggleIsEditing, updateName, onSaveTags]);
+
+  return (
+    <div className={`${className} ui--AddressMenu-buttons`}>
+      {isEditing
+        ? (
+          <Button.Group>
+            <Button
+              icon='times'
+              label={t<string>('Cancel')}
+              onClick={onCancel}
+            />
+            <Button
+              icon='save'
+              label={t<string>('Save')}
+              onClick={onEdit}
+            />
+          </Button.Group>
+        )
+        : (
+          <Button.Group>
+            <Button
+              icon='paper-plane'
+              isDisabled={isEditing}
+              label={t<string>('Send')}
+              onClick={toggleIsTransferOpen}
+            />
+            {!flags.isOwned && !flags.isInContacts && (
+              <Button
+                icon='plus'
+                isDisabled={isEditing}
+                label={t<string>('Save')}
+                onClick={_onUpdateName}
+              />
+            )}
+            {!flags.isOwned && flags.isInContacts && (
+              <Button
+                icon='ban'
+                isDisabled={isEditing}
+                label={t<string>('Remove')}
+                onClick={_onForgetAddress}
+              />
+            )}
+            <Button
+              icon='edit'
+              label={t<string>('Edit')}
+              onClick={onEdit}
+            />
+          </Button.Group>)}
+      {isTransferOpen && (
+        <Transfer
+          key='modal-transfer'
+          onClose={toggleIsTransferOpen}
+          recipientId={recipientId}
+        />
+      )}
+    </div>
+  );
+}
+
+export default React.memo(styled(AccountMenuButtons)`
+  width: 100%;
+
+  .ui--Button-Group {
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    margin-bottom: 0;
+
+    .ui--Button {
+      width: 100%;
+      min-width: max-content;
+    }
+  }
```

Comment:
Not fond of this layout, the buttons here are different to the rest of the UI, i.e. they take up 50% of space each. I would suggest just sticking the the normal group layout.